### ISSUE TITLE
Bumping the date is not working

### ISSUE THREAD

**Issue by DaniBodor** [callowayproject/bump-my-version#275]
* bump-my-version version: 0.29.0
* Python version: 3.11
* Operating System: Ubuntu (WSL)

### Description

I am trying to automatically update the date in one of my files with bump my version as described [here](https://callowayproject.github.io/bump-my-version/howtos/update-a-date/), but it's not getting updated.
How do I tell the system that `now` represents the current date?

### What I Did

I created a test repo that only contains an __init__.py file

```python
__date__ = '2024-12-20'
__version__ = '0.4.0'
```

and a .bumpversion.toml

```yml
[tool.bumpversion]
current_version = "0.4.0"

[[tool.bumpversion.files]]
filename = '__init__.py'
search = "__date__ = '\\d{{4}}-\\d{{2}}-\\d{{2}}'"
replace = "__date__ = '{now:%Y-%m-%d}'"
regex = true

[[tool.bumpversion.files]]
filename = '__init__.py'
```

then I run 
```bash
 bump-my-version bump minor
```

and the result is that the version gets uopdated, but the date stays the same.

**Comment by DaniBodor** [comment]
Side note:

in the docs, the second time the init file is mentioned on that documentation site, it is rendered as a boldface rather than actual double underscored filename.

**Comment by coordt** [comment]
Unfortunately, I can't reproduce your issue. Can you rerun the command with `-vv` added to increase the logging?

**Comment by DaniBodor** [comment]
I apologize, apparently I had mixed up the quotation marks style between my actual test and the one I put above. Indeed, the one above works for me as well, but if I invert the quotation marks (as it is in my actual repo), then it does not work anymore.

init file:
```python
__date__ = "2022-10-02"
__version__ = "0.5.0"
```

bumpversion settings
```yml
[tool.bumpversion]
current_version = "0.5.0"

[[tool.bumpversion.files]]
filename = '__init__.py'
search = '__date__ = "\\d{{4}}-\\d{{2}}-\\d{{2}}"'
replace = '__date__ = "{now:%Y-%m-%d}"'
regex = true

[[tool.bumpversion.files]]
filename = '__init__.py'
```

Here is the output of `bump-my-version bump minor -vv`

> Starting BumpVersion 0.29.0                                                                                                   
> Reading configuration                                                                                                         
>   Reading config file: /home/XXXXX/test_BMV/.bumpversion.toml                                                     
>   Error when running git describe: fatal: No names found, cannot describe anything.                                           
>                                                                                                                               
>   Parsing current version '0.5.0'                                                                                             
>     Parsing version '0.5.0' using regexp '(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)'                                     
>       Parsed the following values: major=0, minor=5, patch=0                                                                  
>   No setup hooks defined                                                                                                      
>   Attempting to increment part 'minor'                                                                                        
>     Values are now: major=0, minor=6, patch=0                                                                                 
>   Serializing version '<bumpversion.Version:major=0, minor=6, patch=0>'                                                       
>     Using serialization format '{major}.{minor}.{patch}'                                                                      
>     Serialized to '0.6.0'                                                                                                     
>   New version will be '0.6.0'                                                                                                 
>                                                                                                                               
> File __init__.py: replace `__date__ = "\\d{{4}}-\\d{{2}}-\\d{{2}}"` with `__date__ = "{now:%Y-%m-%d}"`                        
>   Serializing the current version                                                                                             
>     Serializing version '<bumpversion.Version:major=0, minor=5, patch=0>'                                                     
>       Using serialization format '{major}.{minor}.{patch}'                                                                    
>       Serialized to '0.5.0'                                                                                                   
>   Serializing the new version                                                                                                 
>     Serializing version '<bumpversion.Version:major=0, minor=6, patch=0>'                                                     
>       Using serialization format '{major}.{minor}.{patch}'                                                                    
>       Serialized to '0.6.0'                                                                                                   
>   Rendering search pattern with context                                                                                       
>     Searching for the regex: '__date__ = "\\d{4}-\\d{2}-\\d{2}"'                                                              
>   Found '0\.5\.0' at line 2: 0.5.0                                                                                            
>   Rendering search pattern with context                                                                                       
>     Searching for the regex: '__date__ = "\\d{4}-\\d{2}-\\d{2}"'                                                              
>   Not changing file __init__.py                                                                                               
>                                                                                                                               
> File __init__.py: replace `{current_version}` with `{new_version}`                                                            
>   Serializing the current version                                                                                             
>     Serializing version '<bumpversion.Version:major=0, minor=5, patch=0>'                                                     
>       Using serialization format '{major}.{minor}.{patch}'                                                                    
>       Serialized to '0.5.0'                                                                                                   
>   Serializing the new version                                                                                                 
>     Serializing version '<bumpversion.Version:major=0, minor=6, patch=0>'                                                     
>       Using serialization format '{major}.{minor}.{patch}'                                                                    
>       Serialized to '0.6.0'                                                                                                   
>   Rendering search pattern with context                                                                                       
>     No RegEx flag detected. Searching for the default pattern: '0\.5\.0'                                                      
>   Found '0\.5\.0' at line 2: 0.5.0                                                                                            
>   Changing file __init__.py:                                                                                                  
>     *** before __init__.py                                                                                                    
>     --- after __init__.py                                                                                                     
>     ***************                                                                                                           
>     *** 1,2 ****                                                                                                              
>       __date__ = "2022-10-02"                                                                                                 
>     ! __version__ = "0.5.0"                                                                                                   
>     --- 1,2 ----                                                                                                              
>       __date__ = "2022-10-02"                                                                                                 
>     ! __version__ = "0.6.0"                                                                                                   
>                                                                                                                               
> Processing config file: /home/XXXXX/test_BMV/.bumpversion.toml                                                    
>   Serializing version '<bumpversion.Version:major=0, minor=5, patch=0>'                                                       
>     Using serialization format '{major}.{minor}.{patch}'                                                                      
>     Serialized to '0.5.0'                                                                                                     
>   Serializing version '<bumpversion.Version:major=0, minor=6, patch=0>'                                                       
>     Using serialization format '{major}.{minor}.{patch}'                                                                      
>     Serialized to '0.6.0'                                                                                                     
>   Rendering search pattern with context                                                                                       
>     No RegEx flag detected. Searching for the default pattern: '0\.5\.0'                                                      
>   Found '0\.5\.0' at line 1: 0.5.0                                                                                            
>   Changing file /home/XXXXX/test_BMV/.bumpversion.toml:tool.bumpversion.current_version:                          
>     *** before /home/XXXXX/test_BMV/.bumpversion.toml:tool.bumpversion.current_version                            
>     --- after /home/XXXXX/test_BMV/.bumpversion.toml:tool.bumpversion.current_version                             
>     ***************                                                                                                           
>     *** 1 ****                                                                                                                
>     ! 0.5.0                                                                                                                   
>     --- 1 ----                                                                                                                
>     ! 0.6.0                                                                                                                   
> Serializing version '<bumpversion.Version:major=0, minor=6, patch=0>'                                                         
>   Using serialization format '{major}.{minor}.{patch}'                                                                        
>   Serialized to '0.6.0'                                                                                                       
> No pre-commit hooks defined                                                                                                   
> Would not commit                                                                                                              
> Would not tag                                                                                                                 
> Serializing version '<bumpversion.Version:major=0, minor=6, patch=0>'                                                         
>   Using serialization format '{major}.{minor}.{patch}'                                                                        
>   Serialized to '0.6.0'                                                                                                       
> No post-commit hooks defined                                                                                                  
> Done.

**Comment by DaniBodor** [comment]
I have found a solution, but it is not pretty. It is to use double quotation marks both to indicated the string as well as (escaped) to indicate the quotation marks in the string (see below)

This is odd to me, as plenty of other settings work fine when single quotes are used (including the `filename` below).


The following works.

init:
```python
__date_single__ = '2022-10-02'
__date_double__ = "2022-10-02"
__version__ = '0.4.0'
```

settings:
```yml
[tool.bumpversion]
current_version = "0.4.0"

[[tool.bumpversion.files]]
filename = '__init__.py'
search = "__date_single__ = '\\d{{4}}-\\d{{2}}-\\d{{2}}'"
replace = "__date_single__ = '{now:%Y-%m-%d}'"
regex = true

[[tool.bumpversion.files]]
filename = '__init__.py'
search = "__date_double__ = \"\\d{{4}}-\\d{{2}}-\\d{{2}}\""
replace = "__date_double__ = \"{now:%Y-%m-%d}\""
regex = true

[[tool.bumpversion.files]]
filename = '__init__.py'
```

**Comment by coordt** [comment]
**TL;DR:** 

Replace

```toml
search = '__date__ = "\\d{{4}}-\\d{{2}}-\\d{{2}}"'
```

with

```toml
search = '__date__ = "\d{{4}}-\d{{2}}-\d{{2}}"'
```

**Reason:**

OK, this is an odd thing about either TOML or Python strings (I'm not sure which). The escaping is different when double quotes are in the string. The slashes inside the double quotes don't need to be escaped.

I'll update the docs about this.

**Comment by DaniBodor** [comment]
Perfect, thanks!!
